### PR TITLE
Mmap fix implemented, related hostdesc fix, removal of dead code

### PR DESCRIPTION
**Pull request by jesings**

### CODE REVIEW

File: src/trusted/service_runtime/nacl_syscall_common.c [hunk 1]
```diff
@@ -741,7 +741,7 @@ int32_t NaClSysOpen(struct NaClAppThread  *natp,
   */
   
   fd_retval = NextFd(nap->cage_id);
-  if(hd) hd->lindfd = fd_retval;
+  if(hd) fd_retval = hd->lindfd = hd->d;
```
Comment (rennergade) [inline src/trusted/service_runtime/nacl_syscall_common.c]: We probably need a good bit of text explaining the lindfd thing now, also probably separate these.

File: src/shared/platform/posix/nacl_host_desc.c [hunk 2]
```diff
@@ -156,10 +156,7 @@ uintptr_t NaClHostDescMap(struct NaClHostDesc *d,
   if (flags & NACL_ABI_MAP_ANONYMOUS) {
     desc = -1;
   } else {
-    desc = d->d;
-  }
-  if(-1 != desc) {
-      desc = GetHostFdFromLindFd(desc, d->cageid);
+    desc = d->lindfd;
```
Comment (rennergade) [inline src/shared/platform/posix/nacl_host_desc.c]: this should be lindfd and not d?

Comment (jesings) [reply src/shared/platform/posix/nacl_host_desc.c]: Yep, this goes directly to lind_mmap

**Review by rennergade — COMMENTED** [review]
Okay, this looks great! Clarifying the userfd definitely helps here.

**Pull request merged**